htlcswitch/mailbox.go

```diff
@@ -564,7 +570,7 @@ func (m *memoryMailBox) AddPacket(pkt *htlcPacket) error {
 	case *lnwire.UpdateFulfillHTLC, *lnwire.UpdateFailHTLC:
 		if _, ok := m.pktIndex[pkt.inKey()]; ok {
 			m.pktCond.L.Unlock()
-			return nil
+			return ErrPacketAlreadyExists
```

Comment:
Does the link need to update its behavior if this happens? Right now (this commit), it doesn't check the return value of `AddPacket`.